tapestry.js

```diff
@@ -48,18 +51,50 @@ jQuery.get(apiUrl + "/tapestries/" + tapestryWpPostId, function(result){
 
     tapestrySlug = dataset.settings.tapestrySlug;
     
-    if (saveProgressToCookie) {
-        // Update dataset with data from cookie (if any)
-        var cookieProgress = Cookies.get("progress-data-"+tapestrySlug);
-        if (cookieProgress !== undefined) {
-            cookieProgress = JSON.parse( cookieProgress );
-            setDatasetProgress(cookieProgress);
-        }
-        // Update H5P Video Settings from cookie (if any)
-        var cookieH5PVideoSettings = Cookies.get("h5p-video-settings");
-        if (cookieH5PVideoSettings !== undefined) {
-            cookieH5PVideoSettings = JSON.parse( cookieH5PVideoSettings );
-            h5pVideoSettings = cookieH5PVideoSettings;
+    if (saveProgress) {
+        // If user is logged in, get progress from database database
+        if (tapestryWpUserId) {
+            var getProgData = {
+                "post_id": tapestryWpPostId,
+            };
+
+            jQuery.get(TAPESTRY_PROGRESS_URL, getProgData, function(result) {
+                if (result && !isEmptyObject(result)) {
+                    setDatasetProgress(JSON.parse(result));
+                    updateViewedProgress(); // update viewed progress because async fetch of dataset
+                }
+            }).fail(function(e) {
+                console.log("Error with retrieving node progress");
```

Comment:
Please use `console.error` for error output

Reply:
Changed from console.logs to console.error